ai_council/core/error_handling.py

```
@@ -0,0 +1,392 @@
+        for exc_type, handler in self._custom_handlers.items():
+            if isinstance(exception, exc_type):
+                return handler(exception)
+        
+        # Determine error type from mapping
+        error_type = self._get_error_type(exception)
+        
+        # Determine log level based on exception severity
+        log_level = self._get_log_level(exception)
+        
+        # Log the error
+        log_message = f"{error_type}: {str(exception)}"
```

Ruff (0.15.1)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)